src/client.cpp

```diff
@@ -214,56 +214,94 @@ bool hasIntMember(const rapidjson::Value& value, const char* name)
+        CurlSession& session = *sessionPtr;
 
-    if (document.Parse(responseBody.c_str()).HasParseError())
-    {
-        LOG(ERROR) << "Error parsing response";
-        return makeError();
-    }
+        const std::string& url = apiRoot_;
+        CURLcode res = session.httpGet(url);
 
-    if (hasStringMember(document, kStrAccountsUrl)
-        &&  hasStringMember(document, kStrUrl)
-        &&  hasStringMember(document, kStrVersion))
-    {
-        accountsUrl_ = document[kStrAccountsUrl].GetString();
+        if (res != CURLE_OK)
+        {
+            LOG(ERROR) << fname << ": GET " << url << " failed. "
+                       << "CURLcode: " << res << ", " << curl_easy_strerror(res);
+            rv = makeNetworkError();
+            break;
+        }
 
-        if (!accountsUrl_.empty()  &&  *accountsUrl_.rbegin() != '/')
-            accountsUrl_.push_back('/');
-    }
-    else
-    {
-        LOG(ERROR) << "Response JSON must contain three std::string members: "
-               << kStrAccountsUrl << ", " << kStrUrl << " and " << kStrVersion;
-        return makeError();
-    }
+        long responseCode = session.getResponseCode();
 
-    return makeSuccess();
+        if (responseCode != 200)
+        {
+            LOG(ERROR) << fname << ": GET " << url << " failed."
+                       << " HTTP response code: " << responseCode;
+            rv = makeError(responseCode, Status::FACILITY_HTTP);
+            break;
+        }
+
+        const std::string& responseBody = session.getResponseBodyAsString();
+
+        rapidjson::Document document;
+
+        if (document.Parse(responseBody.c_str()).HasParseError())
+        {
+            LOG(ERROR) << fname << ": error parsing response: '" << responseBody
+                       << "' to GET " << url;
+            rv = makeError();
+            break;
+        }
+
+        if (hasStringMember(document, kStrAccountsUrl)
+            &&  hasStringMember(document, kStrUrl)
+            &&  hasStringMember(document, kStrVersion))
+        {
+            accountsUrl_ = document[kStrAccountsUrl].GetString();
+
+            if (!accountsUrl_.empty()  &&  *accountsUrl_.rbegin() != '/')
+                accountsUrl_.push_back('/');
+        }
+        else
+        {
+            LOG(ERROR) << fname << ": response JSON must contain three std::string members: "
+                       << kStrAccountsUrl << ", " << kStrUrl << " and " << kStrVersion;
+            LOG(ERROR) << fname << ": response: '" << responseBody << "' to GET " << url;
+            rv = makeError();
+            break;
+        }
+
+        rv = makeSuccess();
+    } while (false);
+
+    if (rv.isError())
+        LOG(ERROR) << fname << ": " << rv;
+
+    return rv;
 }
 
-// TODO change prefix parse for something more suitable
-Status parseAccount(const rapidjson::Value& itemJson, Account& account)
+Status parseAccountJson(const rapidjson::Value& itemJson, Account& account)
 {
+    const char* fname = __FUNCTION__;
```

Comment:
It is said that now `__func__`  is the standard way, not `__FUNCTION__`.
At least it is defined in C99 and C++11.